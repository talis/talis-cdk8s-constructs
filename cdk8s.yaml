language: typescript
imports:
  - k8s@1.30.0
