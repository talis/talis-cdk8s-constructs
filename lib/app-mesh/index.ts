export * from "./ingress-gateway-nlb";
export * from "./app-service";
