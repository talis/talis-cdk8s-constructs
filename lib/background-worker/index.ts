export * from "./background-worker-props";
export * from "./background-worker";
