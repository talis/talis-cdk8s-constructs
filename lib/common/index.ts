export * from "./container-props";
export * from "./env-util";
export * from "./name-util";
export * from "./statefulset-util";
export * from "./value-util";
export * from "./workload-props";
