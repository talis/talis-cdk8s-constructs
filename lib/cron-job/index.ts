export * from "./cron-job-props";
export * from "./cron-job";
