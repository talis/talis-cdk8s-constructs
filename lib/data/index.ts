export * from "./config-map";
export * from "./secret";
export * from "./factory";
