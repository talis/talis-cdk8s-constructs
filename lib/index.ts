export * from "./cron-job";
export * from "./background-worker";
export * from "./data";
export * from "./job";
export * from "./redis";
export * from "./talis-chart";
export * from "./web-service";
