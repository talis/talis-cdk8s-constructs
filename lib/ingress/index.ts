export * from "./ingress-props";
export * from "./ingress";
