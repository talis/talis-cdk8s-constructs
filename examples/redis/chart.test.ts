import { RedisChart } from "./chart";
import { Testing } from "cdk8s";
import { TalisShortRegion, TalisDeploymentEnvironment } from "../../lib";

describe("Redis example", () => {
  beforeEach(() => {
    jest.resetModules();
  });

  test("Snapshot", () => {
    const app = Testing.app();
    const chart = new RedisChart(app, {
      environment: TalisDeploymentEnvironment.DEVELOPMENT,
      region: TalisShortRegion.LOCAL,
      watermark: "test",
    });
    const results = Testing.synth(chart);
    expect(results).toMatchSnapshot();
  });
});
