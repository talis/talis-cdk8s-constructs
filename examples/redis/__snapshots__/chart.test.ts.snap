// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Redis example Snapshot 1`] = `
Array [
  Object {
    "apiVersion": "v1",
    "kind": "Namespace",
    "metadata": Object {
      "labels": Object {
        "app": "example-redis-app",
        "environment": "development",
        "managed-by": "cdk8s",
        "region": "local",
        "service": "example-redis-app-development-local",
      },
      "name": "example-redis-app-test",
      "namespace": "example-redis-app-test",
    },
  },
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "labels": Object {
        "app": "example-redis-app",
        "environment": "development",
        "instance": "redis-example",
        "managed-by": "cdk8s",
        "region": "local",
        "release": "5.0.7",
        "role": "redis",
        "service": "example-redis-app-development-local",
      },
      "name": "redis-example",
      "namespace": "example-redis-app-test",
    },
    "spec": Object {
      "ports": Array [
        Object {
          "port": 6379,
          "protocol": "TCP",
        },
      ],
      "selector": Object {
        "app": "example-redis-app",
        "instance": "redis-example",
        "role": "redis",
      },
    },
  },
  Object {
    "apiVersion": "apps/v1",
    "kind": "StatefulSet",
    "metadata": Object {
      "labels": Object {
        "app": "example-redis-app",
        "environment": "development",
        "instance": "redis-example",
        "managed-by": "cdk8s",
        "region": "local",
        "release": "5.0.7",
        "role": "redis",
        "service": "example-redis-app-development-local",
      },
      "name": "redis-example-sts",
      "namespace": "example-redis-app-test",
    },
    "spec": Object {
      "replicas": 1,
      "selector": Object {
        "matchLabels": Object {
          "app": "example-redis-app",
          "instance": "redis-example",
          "role": "redis",
        },
      },
      "serviceName": "redis-example",
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "app": "example-redis-app",
            "instance": "redis-example",
            "role": "redis",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "command": Array [
                "redis-server",
                "--appendonly",
                "yes",
              ],
              "env": Array [
                Object {
                  "name": "MASTER",
                  "value": "true",
                },
              ],
              "image": "redis:5.0.7",
              "name": "redis",
              "ports": Array [
                Object {
                  "containerPort": 6379,
                },
              ],
              "resources": Object {
                "limits": Object {
                  "cpu": "100m",
                  "memory": "250Mb",
                },
              },
              "volumeMounts": Array [
                Object {
                  "mountPath": "/redis-master-data",
                  "name": "data",
                },
              ],
            },
          ],
          "volumes": Array [
            Object {
              "emptyDir": Object {},
              "name": "data",
            },
          ],
        },
      },
    },
  },
]
`;
